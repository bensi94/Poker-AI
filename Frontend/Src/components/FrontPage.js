import React from 'react';


class FrontPage extends React.Component {
    render() {
        return <h1>FrontPage</h1>;
    }
}

export default FrontPage;
