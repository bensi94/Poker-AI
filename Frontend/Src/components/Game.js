import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSocket } from '../actions/socket';

// Temp const should be replaced when we have multiple tables
const CURRENT_TABLE = 'game';

class Game extends React.Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.props.createSocket(CURRENT_TABLE);
    }

    render() {
        return <h1> Game </h1>;
    }
}

Game.propTypes = {
    createSocket: PropTypes.func.isRequired,
    tokenString: PropTypes.string.isRequired,
    players: PropTypes.arrayOf(PropTypes.shape({
        username: PropTypes.string
    }))
};

const mapStateToProps = (state) => {
    if (state.tables[CURRENT_TABLE]) {
        return {
            players: state.tables[CURRENT_TABLE].players,
            tokenString: `Token ${state.auth.token}`
        };
    } else {
        return {
            tokenString: `Token ${state.auth.token}`
        };
    }
};

const mapDispatchToProps = (dispatch) => ({
    createSocket: (table) => dispatch(createSocket(table))
});


export default connect(mapStateToProps, mapDispatchToProps)(Game);
