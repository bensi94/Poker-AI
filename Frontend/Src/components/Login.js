import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form, Input, Button, Layout, Notification } from 'element-react';
import { push } from 'connected-react-router';

import { clearSignup, clearLogin, login } from '../actions/auth';
import { checkUser } from '../actions/user';


export class Login extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            form: {
                username: '',
                password: ''
            },
            rules: {
                username: [
                    { required: true, message: 'Please input the username', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the username'));
                        } else if (this.props.error) {
                            callback(new Error(this.props.error));
                        } else {
                            callback();
                        }
                    } }
                ],
                password: [
                    { required: true, message: 'Please input the password ', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password'));
                        } else {
                            callback();
                        }
                    } }
                ]
            }
        };

        this.notification = this.notification.bind(this);
    }

    componentDidMount() {
        if (this.props.username) {
            this.notification(this.props.username);
            this.props.clearSignup();
        }
    }

    componentDidUpdate() {
        if (this.props.token && this.props.isAuthenticated) {
            this.props.redirectGame();
        } else if (this.props.error) {
            this.refs.form.validate();
            this.props.clearLogin();
        }
    }


    handleSubmit(e) {
        e.preventDefault();

        this.refs.form.validate((valid) => {
            if (valid) {
                this.props.login(this.state.form).then(() => {
                    this.props.checkUser().then(() => {
                        this.props.redirectGame();
                    });
                });
            } else {
                return false;
            }
        });
    }

    handleReset(e) {
        e.preventDefault();

        this.refs.form.resetFields();
    }

    onChange(key, value) {
        this.setState({
            form: Object.assign({}, this.state.form, { [key]: value })
        });
    }

    notification(name) {
        // eslint-disable-next-line new-cap
        Notification({
            title: 'User created',
            message: 'Created new user: ' + name,
            type: 'success'
        });
    }

    render() {
        return (
            <Layout.Row type="flex" justify="center">
                <Form ref="form"
                    model={this.state.form}
                    rules={this.state.rules}
                    labelWidth="100"
                    className="signup-login-form"
                >
                    <h2 className="signup-login-title">Login</h2>
                    <Form.Item label="Username" prop="username">
                        <Input type="text"
                            value={this.state.form.username}
                            onChange={this.onChange.bind(this, 'username')}
                        />
                    </Form.Item>
                    <Form.Item label="Pasword" prop="password">
                        <Input type="password"
                            value={this.state.form.password}
                            onChange={this.onChange.bind(this, 'password')}
                        />
                    </Form.Item>
                    <Form.Item>
                        <Button type="primary" onClick={this.handleSubmit.bind(this)}>Login</Button>
                        <Button onClick={this.handleReset.bind(this)}>Reset</Button>
                    </Form.Item>
                </Form>
            </Layout.Row>
        );
    }
}

Login.propTypes = {
    clearSignup: PropTypes.func.isRequired,
    clearLogin: PropTypes.func.isRequired,
    login: PropTypes.func.isRequired,
    redirectGame: PropTypes.func.isRequired,
    checkUser: PropTypes.func.isRequired,
    username: PropTypes.string,
    token: PropTypes.string,
    error: PropTypes.string,
    isAuthenticated: PropTypes.bool
};

const mapStateToProps = (state) => {
    return {
        username: state.auth.username,
        token: state.auth.token,
        error: state.auth.error,
        isAuthenticated: state.user.isAuthenticated
    };
};

const mapDispatchToProps = (dispatch) => ({
    clearSignup: () => dispatch(clearSignup()),
    clearLogin: () => dispatch(clearLogin()),
    login: (user) => dispatch(login(user)),
    redirectGame: () => dispatch(push('/game')),
    checkUser: () => dispatch(checkUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
