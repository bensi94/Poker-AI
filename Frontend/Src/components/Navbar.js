// Dependencies
import React from 'react';
import { Menu } from 'element-react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { clearLogin } from '../actions/auth';
import { checkUser } from '../actions/user';


export class Navbar extends React.Component {
    constructor(props) {
        super(props);
        this.updateIsActive = this.updateIsActive.bind(this);
    }

    updateIsActive(match) {
        if (match && match.url === this.props.active) {
            return true;
        }
        return false;
    }

    render() {
        if (this.props.isAuthenticated) {
            return (
                <Menu className="navBar" mode="horizontal" theme="dark">
                    <NavLink exact to="/table/1"
                        className="navlink"
                        activeClassName="active-navlink"
                        isActive={this.updateIsActive}
                    >
                        <Menu.Item index="1">
                            Rounders Poker
                        </Menu.Item>
                    </NavLink>
                    <NavLink exact to="/"
                        className="navlink"
                        activeClassName="active-navlink"
                        isActive={this.updateIsActive}
                        onClick={() => {
                            this.props.logOut();
                            this.props.checkUser();
                        }}
                    >
                        <Menu.Item className="right-item" index="2">
                            Log out
                        </Menu.Item>
                    </NavLink>
                </Menu>
            );
        } else {
            return (
                <Menu className="navBar" mode="horizontal" theme="dark">
                    <NavLink exact to="/"
                        className="navlink"
                        activeClassName="active-navlink"
                        isActive={this.updateIsActive}
                    >
                        <Menu.Item index="1">
                            Rounders Poker
                        </Menu.Item>
                    </NavLink>
                    <NavLink exact to="/login"
                        className="navlink"
                        activeClassName="active-navlink"
                        isActive={this.updateIsActive}
                    >
                        <Menu.Item className="right-item" index="2">
                            Login
                        </Menu.Item>
                    </NavLink>
                    <NavLink exact to="/signup"
                        className="navlink"
                        activeClassName="active-navlink"
                        isActive={this.updateIsActive}
                    >
                        <Menu.Item className="right-item" index="3">
                            Sign up
                        </Menu.Item>
                    </NavLink>
                </Menu>
            );
        }
    }
}

Navbar.propTypes = {
    active: PropTypes.string.isRequired,
    isAuthenticated: PropTypes.bool.isRequired,
    logOut: PropTypes.func.isRequired,
    checkUser: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
    return {
        active: state.router.location.pathname,
        isAuthenticated: state.user.isAuthenticated
    };
};

const mapDispatchToProps = (dispatch) => ({
    logOut: () => dispatch(clearLogin()),
    checkUser: () => dispatch(checkUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
