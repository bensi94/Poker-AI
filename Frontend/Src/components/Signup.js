import React from 'react';
import { Form, Input, Button, Layout } from 'element-react';


class Signup extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            form: {
                username: '',
                password: '',
                confirmPassword: ''
            },
            rules: {
                username: [
                    { required: true, message: 'Please input the username', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the username'));
                        } else {
                            callback();
                        }
                    } }
                ],
                password: [
                    { required: true, message: 'Please input the password', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password'));
                        } else if (value.length < 5) {
                            callback(new Error('Password must be at least 5 characters'));
                        } else {
                            if (this.state.form.confirmPassword !== '') {
                                this.refs.form.validateField('confirmPassword');
                            }
                            callback();
                        }
                    } }
                ],
                confirmPassword: [
                    { required: true, message: 'Please input the password again', trigger: 'blur' },
                    { validator: (rule, value, callback) => {
                        if (value === '') {
                            callback(new Error('Please input the password again'));
                        } else if (value !== this.state.form.password) {
                            callback(new Error('The passwords do not match'));
                        } else {
                            callback();
                        }
                    } }
                ]
            }
        };
    }

    handleSubmit(e) {
        e.preventDefault();

        this.refs.form.validate((valid) => {
            if (valid) {
                console.log('isValid');
            } else {
                console.log('invalid');
                return false;
            }
        });
    }

    handleReset(e) {
        e.preventDefault();

        this.refs.form.resetFields();
    }

    onChange(key, value) {
        this.setState({
            form: Object.assign({}, this.state.form, { [key]: value })
        });
    }

    render() {
        return (
            <Layout.Row type="flex" justify="center">
                <Form ref="form"
                    model={this.state.form}
                    rules={this.state.rules}
                    labelWidth="100"
                    className="signup-login-form"
                >
                    <h2 className="signup-login-title">Sign up</h2>
                    <Form.Item label="Username" prop="username">
                        <Input type="text"
                            value={this.state.form.username}
                            onChange={this.onChange.bind(this, 'username')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item label="Pasword" prop="password">
                        <Input type="password"
                            value={this.state.form.password}
                            onChange={this.onChange.bind(this, 'password')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item label="Confirm password" prop="confirmPassword">
                        <Input type="password"
                            value={this.state.form.confirmPassword}
                            onChange={this.onChange.bind(this, 'confirmPassword')}
                            autoComplete="off"
                        />
                    </Form.Item>
                    <Form.Item>
                        <Button type="primary" onClick={this.handleSubmit.bind(this)}>Sign up</Button>
                        <Button onClick={this.handleReset.bind(this)}>Reset</Button>
                    </Form.Item>
                </Form>
            </Layout.Row>
        );
    }
}

export default Signup;
