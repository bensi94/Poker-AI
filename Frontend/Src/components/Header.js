// Dependencies
import React from 'react';
import { Route, Switch } from 'react-router';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Loading } from 'element-react';


// Project files
import FrontPage from './FrontPage';
import Login from './Login';
import Signup from './Signup';
import Navbar from './Navbar';
import Game from './Game';
import PrivateRoute from '../routers/PrivateRoute';
import PublicRoute from '../routers/PublicRoute';
import { checkUser } from '../actions/user';


export class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        let page;
        if (this.props.isAuthenticated === undefined ||
            (this.props.isAuthenticated && !this.props.token)) {
            page = <Loading fullscreen/>;
            this.props.checkUser();
        } else {
            page = (
                <React.Fragment>
                    <Navbar />
                    <Switch>
                        <Route exact path="/" component={FrontPage} />
                        <PublicRoute exact path="/login" component={Login} />
                        <PublicRoute exact path="/signup" component={Signup} />
                        <PrivateRoute exact path="/table/:tableID" component={Game} />
                        <Route component={FrontPage} />
                    </Switch>
                </React.Fragment>
            );
        }
        return (
            { ...page }
        );
    }
}

Header.propTypes = {
    username: PropTypes.string,
    name: PropTypes.string,
    isAuthenticated: PropTypes.bool,
    token: PropTypes.string,
    error: PropTypes.shape({
        detail: PropTypes.string
    }),
    checkUser: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
    return {
        username: state.user.username,
        name: state.user.name,
        isAuthenticated: state.user.isAuthenticated,
        token: state.auth.token,
        error: state.user.error
    };
};

const mapDispatchToProps = (dispatch) => ({
    checkUser: () => dispatch(checkUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
