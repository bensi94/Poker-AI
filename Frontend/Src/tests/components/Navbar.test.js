import React from 'react';
import { shallow } from 'enzyme';

import { Navbar } from '../../components/Navbar';


describe('Navbar Test suite', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Navbar
            active={''}
            isAuthenticated={false}
            logOut={jest.fn()}
            checkUser={jest.fn()}
        />);
    });

    it('Should render Navabar(snapshot) correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should have router link to each menu on unauthenticated', () => {
        expect(wrapper.find('NavLink')).toHaveLength(3);
    });

    it('Should have router link to each menu on unauthenticated', () => {
        wrapper = shallow(<Navbar
            active={''}
            isAuthenticated
            logOut={jest.fn()}
            checkUser={jest.fn()}
        />);
        expect(wrapper.find('NavLink')).toHaveLength(2);
    });
});
