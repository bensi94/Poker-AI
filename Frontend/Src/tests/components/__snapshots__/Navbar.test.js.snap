// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Test suite Should render Navabar(snapshot) correctly 1`] = `
<Menu
  className="navBar"
  menuTrigger="hover"
  mode="horizontal"
  theme="dark"
>
  <NavLink
    activeClassName="active-navlink"
    className="navlink"
    exact={true}
    isActive={[Function]}
    to="/"
  >
    <MenuItem
      index="1"
    >
      Rounders Poker
    </MenuItem>
  </NavLink>
  <NavLink
    activeClassName="active-navlink"
    className="navlink"
    exact={true}
    isActive={[Function]}
    to="/login"
  >
    <MenuItem
      className="right-item"
      index="2"
    >
      Login
    </MenuItem>
  </NavLink>
  <NavLink
    activeClassName="active-navlink"
    className="navlink"
    exact={true}
    isActive={[Function]}
    to="/signup"
  >
    <MenuItem
      className="right-item"
      index="3"
    >
      Sign up
    </MenuItem>
  </NavLink>
</Menu>
`;
